src/main/java/hudson/plugins/git/browser/Gitiles.java

```diff
@@ -79,21 +77,11 @@ public FormValidation doCheckRepoUrl(@AncestorInPath Item project, @QueryParamet
                 throws IOException, ServletException, URISyntaxException {
 
             String cleanUrl = Util.fixEmptyAndTrim(repoUrl);
-
-            if (cleanUrl == null) {
-                return FormValidation.ok();
-            }
-
-            if (project == null || !project.hasPermission(Item.CONFIGURE)) {
-                return FormValidation.ok();
-            }
-
-            if (cleanUrl.contains("$")) {
-                // set by variable, can't validate
+            if(initialChecksAndReturnOk(project, cleanUrl)){
                 return FormValidation.ok();
             }
             FormValidation response;
-            if (checkURIFormat(cleanUrl)) {
+            if (checkURIFormat(cleanUrl, "gerrit")) {
```

Comment:
Can't assume that the Gerrit server includes "gerrit" in its hostname.

Reply:
Agreed, implementation has been changed.